docs/Coding-Standards/Clean-Code.md

```diff
@@ -0,0 +1,322 @@
+
+Try to limit the number of arguments in a function. When possible, try to avoid two or more arguments. More arguments make a function harder to read and understand. It is also more difficult to test, since they create the need to write test cases for every combination of arguments.
+
+❌
+
+```js
+// add bad example
+```
+
+✅
+
+```js
+// add good example
+```
+
+#### Use object destructuring for functions that require multiple arguments
+
+Sometimes you can’t avoid having multiple function arguments. When this is the case it is recommended to use 1 object argument with object destructuring syntax. This has a few advantages over multiple arguments:
+
+- When someone looks at the function signature, it's immediately clear what properties are being used.
+- It can be used to simulate named parameters.
+- Destructuring also clones the specified primitive values of the argument object passed into the function. This can help prevent side effects. Note: objects and arrays that are destructured from the argument object are NOT cloned.
+- Linters can warn you about unused properties, which would be impossible without destructuring.
+- The order of object properties does not matter whereas the order of function arguments does matter. Especially useful when some parameters are optional and you can leave them out of the object rather than pass `undefined` or `null` as an argument when the optional argument is not the last argument.
+
+❌
+
+```js
+function createMenu(title, body, buttonText, cancellable) {
+  // ...
+}
+
+createMenu('Foo', 'Bar', 'Baz', true);
+```
+
+✅
+
+```js
+function createMenu({ title, body, buttonText, cancellable }) {
+  // ...
+}
+
+createMenu({
+  cancellable: true
+  buttonText: "Baz",
+  body: "Bar",
+  title: "Foo",
+});
+```
+
+#### Avoid flag arguments
+
+A flag argument is a kind of function argument that tells the function to carry out a different operation depending on its value. By design, they are doing more than one thing. If the flag is true, then run a block. Otherwise, another. Instead, you should split the function into two different functions.
+
+❌
+
+```js
+const makeBooking = (customer, isPremium) => {
+  if (isPremium) {
+    // logic for premium booking
+  } else {
+    // logic for regular booking
+  }
+};
+```
+
+✅
+
+```js
+const makeRegularBooking = customer => {
+  // logic for regular booking
+};
+
+const makePremiumBooking = customer => {
+  // logic for premium booking
+};
+```
+
+#### Write pure functions
+
+A function is called a pure function if it always returns the same result when given the same argument values and it has no side effects like modifying an argument or a global variable.
+
+Pure functions are also extremely independent meaning they are easy to move around, refactor, and reorganise in your code, making your applications more flexible and adaptable to future changes. It also makes them immune to bugs that have to do with shared mutable state.
+
+❌
+
+```js
+// add bad example
```

Comment:
// Modifies its argument
const getLastElement = data => data.pop();

// Accesses external variables, so not guaranteed to return the same value for the same argument
const getLastElement = () => data[data.length-1];